src/SFML/System/Time.cpp

```diff
@@ -44,7 +44,7 @@ m_microseconds(0)
 ////////////////////////////////////////////////////////////
 float Time::asSeconds() const
 {
-    return m_microseconds / 1000000.f;
+    return static_cast<float>(static_cast<double>(m_microseconds) / 1000000.0);
```

Comment:
Did this cause a warning in the first place?

Also, is double division and then cast to float more precise than float division?

Reply:
Yes, because Uint64 and double have [different precision](https://stackoverflow.com/questions/30052710/why-double-can-store-bigger-numbers-than-unsigned-long-long)  if you look at the IEEE spec.

I'd assume so, if you have a very big integer and cast it to float, you will lose a lot of precision, while with a double, you'll have more precision.

But I'm open to suggestions, as I don't have too much experience with type conversions at this level. 😄

Reply:
Maybe we should test it for some very low and very high values. 
Increased precision would be a nice side effect of this refactoring 🙂